app/src/common/shared/com/igalia/wolvic/browser/api/WRuntimeSettings.java

```diff
@@ -0,0 +1,666 @@
+package com.igalia.wolvic.browser.api;
+
+import android.app.Service;
+import android.os.Build;
+import android.os.Bundle;
+
+import androidx.annotation.AnyThread;
+import androidx.annotation.IntDef;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+public class WRuntimeSettings {
+    /** A light theme for web content is preferred. */
+    public static final int COLOR_SCHEME_LIGHT = 0;
+    /** A dark theme for web content is preferred. */
+    public static final int COLOR_SCHEME_DARK = 1;
+    /** The preferred color scheme will be based on system settings. */
+    public static final int COLOR_SCHEME_SYSTEM = -1;
+
+    @Retention(RetentionPolicy.SOURCE)
+    @IntDef({COLOR_SCHEME_LIGHT, COLOR_SCHEME_DARK, COLOR_SCHEME_SYSTEM})
+            public @interface ColorScheme {}
+
+    @Retention(RetentionPolicy.SOURCE)
+    @IntDef({ALLOW_ALL, HTTPS_ONLY_PRIVATE, HTTPS_ONLY})
+            public @interface HttpsOnlyMode {}
+
+    /** Allow all insecure connections */
+    public static final int ALLOW_ALL = 0;
+    /** Allow insecure connections in normal browsing, but only HTTPS in private browsing. */
+    public static final int HTTPS_ONLY_PRIVATE = 1;
+    /** Only allow HTTPS connections. */
+    public static final int HTTPS_ONLY = 2;
+
+
+    /** Settings builder used to construct the settings object. */
+    @AnyThread
+    public static final class Builder {
+
+        private WRuntimeSettings mSettings = new WRuntimeSettings();
+
+        private WRuntimeSettings getSettings() {
+            return mSettings;
+        }
+
+        public WRuntimeSettings build() {
+            return mSettings;
+        }
+        /**
+         * Set the custom Gecko process arguments.
```

Comment:
Nit: replace Gecko in the docs in this method and the following ones by something like Web Engine for example